jbpm-benchmarks/jbpm-performance-tests/src/main/java/org/jbpm/test/performance/jbpm/JBPMController.java

```diff
@@ -104,26 +99,33 @@ public void clear() {
     public void tearDown() {
         clear();
         if (persistence) {
-            if (emf != null) {
-                emf.close();
-                emf = null;
-                EntityManagerFactoryManager.get().clear();
-
-            }
-            if (ds != null) {
-                ds.close();
-                ds = null;
-            }
             try {
+                if (emf != null) {
+                    emf.close();
+                    emf = null;
+                    EntityManagerFactoryManager.get().clear();
+
+                }
+                if (ds != null) {
+                    ds.close();
+                    ds = null;
+                }
+
                 InitialContext context = new InitialContext();
-                UserTransaction ut = (UserTransaction) context.lookup(
-                        JtaTransactionManager.DEFAULT_USER_TRANSACTION_NAME);
+                UserTransaction ut = (UserTransaction) context.lookup(JtaTransactionManager.DEFAULT_USER_TRANSACTION_NAME);
                 if (ut.getStatus() != Status.STATUS_NO_TRANSACTION) {
                     ut.setRollbackOnly();
                     ut.rollback();
                 }
             } catch (Exception e) {
-                // do nothing
+                e.printStackTrace();
```

Comment:
Let's use logging, please

Reply:
Or throw an exception? Would it be ok that teardown ends with an Exception?

Reply:
I changed it for logging. I guess the exception cannot be rethrown because otherwise the code wouldn't reach the reporting phase, so I would get no results even though the run was fine but there was for example something with network/DB/lab which prevented it from closing datasource etc.